### ISSUE TITLE
[관리자] 도메인 실패 metric 설계 및 관측

### ISSUE THREAD

**Issue by Vorschlag-bit** [Bockerl/Snail_Backend#70]
## 💡어떤 기능인가요?
> 도메인 별, 해당 도메인의 메서드 별 실패 실시간 데이터를 관측하기 위한 기능을 구현할 예정입니다.

## ✅ 작업 상세 내용

- [x] DomainFailEvent 설계
- [x] 각 도메인별 onFailure에 비동기로 eventpublish
- [x] Kafka Stream으로 domain-fail-event에서 groupBy()로 domain, path별 상태 객체 생성 및 저장
- [x] 상태 객체를 사용자 정의 metric으로 변환 후, prometheus에 저장 후 grafana로 조회
